### ISSUE TITLE
The `--debug` option aborts with gdbserver >= 8.3

### ISSUE THREAD

**Issue by MrAnno** [Snaipe/Criterion#301]
gdbserver requires the first parameter to be `HOST:PORT`.
In gdbserver 8.3, the hostname is actually resolved with `getaddrinfo()`.
Currently, BoxFort uses the string `tcp` as a hostname, which results in a SIGABRT because of the unsuccessful name resolution:

```
[----] Criterion v2.3.3
[====] Running 4 tests from date:
tcp:1234: cannot resolve name: No address associated with hostname
Exiting
criterion: Could not spawn test instance: Protocol error
fish: “modules/date/tests/test_date --…” terminated by signal SIGABRT (Abort)
```

**Comment by MrAnno** [comment]
@Snaipe Thanks for merging diacritic/BoxFort#22.

This issue affects distros where gdbserver is quite new, for example, Arch Linux.

Are you planning to bump the BoxFort subproject hash in Criterion and publish a new release?
If not, I can apply a simple patch on the Arch Linux AUR package easily.

**Comment by Snaipe** [comment]
Please apply a patch on the AUR for now -- I'll release 2.3.4 with that in it, but I'm usually waiting a week or two before calling a new release in case further issues are discovered

**Comment by bno1** [comment]
Any update on this? Ubuntu 20 ships with gdbserver 9.1

**Comment by MrAnno** [comment]
@Snaipe Could you release a v2.3.x version with this patch applied, please?
More and more distributions started to use  `gdbserver > 8.3`, and debugging became problematic without this fix.

**Comment by gaborznagy** [comment]
Hi @Snaipe, I've just ran into this on Ubuntu 20.04 as well.

After I've applied the patch from https://github.com/Snaipe/BoxFort/pull/22 onto release 2.3.3 tar.bz2 and it fixed the problem!

**Comment by MrAnno** [comment]
@Snaipe Pretty please

Can I somehow help you with this?

**Comment by borjasotomayor** [comment]
I ran into this same issue on Ubuntu 20.04 and, in case it helps folks, a quick-and-dirty workaround (to at least be able to debug your tests) is to make "tcp" an alias for "localhost" by running this before the test executable:

```
TMP_HOSTS=$(mktemp); echo 'tcp localhost' > $TMP_HOSTS; export HOSTALIASES=$TMP_HOSTS
```

(note how this doesn't require modifying `/etc/hosts`; it just creates a per-user alias in a temp file)

**Comment by MrAnno** [comment]
@Snaipe could you have a look at this, please?

**Comment by MrAnno** [comment]
This is fixed by the upcoming v2.4.0.
